test_aodndata/gsla/test_gsla.py

```diff
@@ -0,0 +1,149 @@
+OLDER_CREATION_DATE_NC = 'IMOS_OceanCurrent_HV_20180101T000000Z_GSLA_FV02_DM00_C-20100101T000000Z.nc.zip'
+
+NC_FILE = "IMOS_OceanCurrent_HV_20180101T000000Z_GSLA_FV02_DM00_C-20181231T225959Z.nc"
+
+
+class TestGslaHandler(HandlerTestCase):
+    def setUp(self):
+        self.handler_class = GslaHandler
+        super(TestGslaHandler, self).setUp()
+
+    def test_push_unzipped_nc(self):
+        self.run_handler_with_exception(InvalidFileFormatError, NC_FILE)
+
+    def test_good_nc(self):
+        dest_path = GslaHandler.dest_path(GOOD_NC_DM00)
+        expected_path = os.path.join(GSLA_PREFIX_PATH, "DM00/1993", GOOD_NC_DM00)
+        self.assertEqual(dest_path, expected_path)
+
+        dest_path = GslaHandler.dest_path(GOOD_NC_NRT00)
+        expected_path = os.path.join(GSLA_PREFIX_PATH, "NRT00/2018", GOOD_NC_NRT00)
+        self.assertEqual(dest_path, expected_path)
+
+    def test_get_fields(self):
+        fields = get_fields_from_filename(GOOD_NC_DM00, GSLA_REGEX)
+
+        self.assertEqual(fields['nc_time_cov_start'], '19930101T000000Z')
+        self.assertEqual(fields['creation_date'], '20130913T082343Z')
+        self.assertEqual(fields['product_type'], 'DM00')
+
+    def test_get_creation_date(self):
+        creation_date = get_creation_date(GOOD_NC_DM00, NC_CREATION_DATE)
+        self.assertEqual(creation_date, datetime(2013, 9, 13, 8, 23, 43))
+
+    def test_setup_upload_location_push_newer_file(self):
+        """
+        Test case: Check creation date of new *.nc.gz is newer that one is already on storage
+                   HARVEST_ONLY the content of the *.gz
+                   PUBLISH_ONLY the *.nc.gz
+        """
+        # create some PipelineFiles to represent the existing files on 'S3'
+        preexisting_files = PipelineFileCollection()
+
+        existing_file = PipelineFile(PREV_NC_STORAGE, dest_path=os.path.join(
+            'IMOS/OceanCurrent/GSLA/DM00/2018/',
+            os.path.basename(PREV_NC_STORAGE)))
+
+        preexisting_files.update([existing_file])
+
+        # set the files to UPLOAD_ONLY
+        preexisting_files.set_publish_types(PipelineFilePublishType.UPLOAD_ONLY)
+
+        # upload the 'preexisting_files' collection to the unit test's temporary upload location
+        broker = get_storage_broker(self.config.pipeline_config['global']['upload_uri'])
+        broker.upload(preexisting_files)
+
+        # run the handler
+        handler = self.run_handler(NEWER_CREATION_DATE_NC)
+
+        nc_file = handler.file_collection.filter_by_attribute_id('file_type', FileType.NETCDF)[0]
+        self.assertEqual(nc_file.publish_type, PipelineFilePublishType.HARVEST_ONLY)
+
+        gz_nc_file = handler.file_collection.filter_by_attribute_id('file_type', FileType.ZIP)[0]
+        self.assertEqual(gz_nc_file.publish_type, PipelineFilePublishType.UPLOAD_ONLY)
+
+    @patch('aodndata.gsla.handler.GSLA_PREFIX_PATH', '')
```

Comment:
@lwgordonimos 
I want to change to global variable GSLA_PREFIX_PATH to something empty. Like this I can check that the handler doesn't delete any file not belonging to it's folder of choice (ie IMOS/OceanCurrent ...)
unfortunately the patch method doesn't seem to do what I want

Reply:
I think that feature would be better in core, because it's a generically useful feature for all handlers really.

For example, an optional handler parameter which is a a prefix for which to allow deletes or maybe even deletes  *and* additions.

For example, something like:

```python
handler = MyHandler(incoming_file, prefix_constraint='IMOS/OceanCurrent')
```
Then, as a pre-publishing validation action, we can throw an error if any files have a dest_path which lies outside the allowed prefix.

Reply:
We definitely need some kind of safety net in the pipeline doing deletions. Adding this option is quite a sensible way to achieve it.